lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +300,76 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, key:, interval:, **options)
+    raise ArgumentError, "Expected <key> to be present" unless key.present?
+    raise ArgumentError, "Expected <interval> to be present" unless interval.present?
+
+    invalid_options = options.keys - PARTMAN_CREATE_PARENT_OPTIONS
+
+    raise ArgumentError, "Unrecognized optional argument(s): #{invalid_options}" unless invalid_options.empty?
+
+    if ActiveRecord::Base.connection.postgresql_version < 10_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Native partitioning not supported on Postgres databases before version 10"
+    end
+
+    options[:template_table] = _fully_qualified_name(options[:template_table]) if options[:template_table].present?
+
+    options = options.merge(
+      parent_table: _fully_qualified_name(table),
+      control: key,
+      type: "native",
+      interval: interval,
+    ).compact
+
+    create_parent_sql = options.map { |k, v| "p_#{k} := '#{v}'" }.join(", ")
```

Comment:
I don't love manually munging SQL, but we do it elsewhere. Nonetheless, we should probably use the PG gem's string escaping code at least.